hmt_escrow/job.py

```diff
@@ -1432,9 +1436,10 @@ def _create_escrow(self,
                 escrow_created = True
                 break
             except Exception as e:
-                LOG.error(
-                    f"Contract creation failed with {gas_payer} and {gas_payer_priv} due to {e}. Raffling new ones..."
-                )
+                if gas_payer == self.multi_credentials[-1][0]:
```

Comment:
perhaps here you should just use an iterator in the loop?  I fear this check might be a bit fragile

Reply:
yea, good idea

Reply:
going to use already defined boolean fields to perform this, more elegant.